cmd/go-tpc/tpch.go

```diff
@@ -26,19 +27,26 @@ var queryTuningVars = []struct {
 	{"tidb_prefer_broadcast_join_by_exchange_data_size", "ON"},
 }
 
-func appendQueryTuningVarsToConnParams() {
-	for _, v := range queryTuningVars {
-		if !strings.Contains(connParams, v.name) {
-			connParams = fmt.Sprintf("%s&%s=%s", connParams, v.name, v.value)
-		}
+// isSysVarSupported determines if a system variable is supported in given TiDB version
+// INFO: Should be optimized if some vars are changed in the future.
+func isSysVarSupported(ver util.SemVersion, sysVar string) bool {
+	if ver.Major > 7 {
+		return true
+	}
+	if ver.Major < 7 {
+		return false
+	}
+	if ver.Minor > 1 {
+		return true
+	}
+	if ver.Minor < 1 {
+		return false
 	}
+
+	return ver.Patch >= 0
```

Comment:
I think a utility function `CompareSemVer(a, b SemVersion)` might be very helpful. With this in place, we can instead simplify this function to `return CompareSemVer(ver, SemVersion{7, 1, 0}) >= 0`.